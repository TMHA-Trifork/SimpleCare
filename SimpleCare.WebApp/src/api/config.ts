export const API_CONFIG = {
  baseUrl: 'http://localhost:5034'
}

export function getApiUrl(path: string): string {
  return `${API_CONFIG.baseUrl}${path}`
}
