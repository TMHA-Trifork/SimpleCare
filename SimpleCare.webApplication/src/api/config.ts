export const API_CONFIG = {
  baseUrl: 'https://localhost:7023'
}

export function getApiUrl(path: string): string {
  return `${API_CONFIG.baseUrl}${path}`;
}
