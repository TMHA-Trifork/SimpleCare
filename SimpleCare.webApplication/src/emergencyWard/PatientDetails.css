.patient-details-container {
  padding: 20px;
  max-width: 600px;
  margin: 50px auto;
  background: #fff;
  border-radius: 10px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  font-family: Arial, sans-serif;
}

.patient-details-container h1 {
  font-size: 2rem;
  margin-bottom: 20px;
  color: #333;
}

.patient-details-title {
  font-size: 2.5rem;
  color: #333;
  text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.2);
  text-align: center; /* Aligns the title to match Emergency Ward */
}

.patient-details-container p {
  font-size: 1.2rem;
  margin: 10px 0;
  color: #555;
}

.back-button {
  margin-top: 20px;
  padding: 10px 15px;
  font-size: 1rem;
  color: #fff;
  background-color: #2370c2;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  transition: background-color 0.3s;
}

.back-button:hover {
  background-color: #0056b3;
}
