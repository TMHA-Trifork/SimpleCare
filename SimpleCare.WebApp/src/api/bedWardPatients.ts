import { components } from '../types/api-types'
import { getApiUrl } from './config'

export type BedWardPatientListItem = components['schemas']['BedWardPatientListItem']
export type BedWardPatient = components['schemas']['BedWardPatient']
export type PatientAdmission = components['schemas']['PatientAdmission']

export async function fetchBedWardPatients(): Promise<BedWardPatientListItem[]> {
  const response = await fetch(getApiUrl('/api/bed-ward/patients'))
  if (!response.ok) {
    throw new Error('Network response was not ok')
  }
  return response.json()
}

export async function fetchBedWardPatient(patientId: string): Promise<BedWardPatient> {
  const response = await fetch(getApiUrl(`/api/bed-ward/patients/${patientId}`))
  if (!response.ok) {
    throw new Error('Network response was not ok')
  }
  return response.json()
}

export async function admitPatient(admission: PatientAdmission): Promise<void> {
  const response = await fetch(getApiUrl('/api/bed-ward/patients/admit'), {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(admission)
  })
  if (!response.ok) {
    throw new Error('Network response was not ok')
  }
}
