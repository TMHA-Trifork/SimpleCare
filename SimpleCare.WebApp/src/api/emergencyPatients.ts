import { components } from "../types/api-types";
import { getApiUrl } from "./config";

export type EmergencyPatientListItem =
  components["schemas"]["EmergencyPatientListItem"];
export type EmergencyPatient = components["schemas"]["EmergencyPatient"];
export type EmergencyRegistration =
  components["schemas"]["EmergencyRegistration"];
export type TransferRequest = components["schemas"]["TransferRequest"];

export async function fetchEmergencyPatients(
  status?: components["schemas"]["EmergencyPatientStatus"][]
): Promise<EmergencyPatientListItem[]> {
  const url = new URL(getApiUrl("/api/emergency-wards/patients"));
  if (status && status.length > 0) {
    status.forEach((s) => url.searchParams.append("status", s));
  }
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
  return response.json();
}

export async function fetchEmergencyPatient(
  patientId: string
): Promise<EmergencyPatient> {
  const response = await fetch(
    getApiUrl(`/api/emergency-wards/patients/${patientId}`)
  );
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
  return response.json();
}

export async function registerEmergencyPatient(
  registration: EmergencyRegistration
): Promise<void> {
  const response = await fetch(
    getApiUrl("/api/emergency-wards/patients/register"),
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(registration),
    }
  );
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
}

export async function transferEmergencyPatient(
  transfer: TransferRequest
): Promise<void> {
  const response = await fetch(
    getApiUrl("/api/emergency-wards/patients/transfer"),
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(transfer),
    }
  );
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
}
