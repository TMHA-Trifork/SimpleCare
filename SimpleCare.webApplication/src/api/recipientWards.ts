import { getApiUrl } from "./config";

export type RecipientWard = {
  id: string;
  name: string;
};

export async function fetchRecipientWards(): Promise<RecipientWard[]> {
  const response = await fetch(
    getApiUrl("/api/emergency-wards/recipient-wards")
  );
  if (!response.ok) {
    throw new Error("Network response was not ok");
  }
  return response.json();
}
