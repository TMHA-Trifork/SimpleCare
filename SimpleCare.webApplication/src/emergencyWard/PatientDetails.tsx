import React, { useEffect, useState } from "react";
import { useParams, useNavigate } from "react-router-dom";
import {
  EmergencyPatient,
  fetchEmergencyPatient,
} from "../api/emergencyPatients";
import "./PatientDetails.css";

const PatientDetails: React.FC = () => {
  const { id } = useParams<{ id: string }>();
  const navigate = useNavigate();
  const [patient, setPatient] = useState<EmergencyPatient | null>(null);

  useEffect(() => {
    const loadPatient = async () => {
      try {
        if (id) {
          const data = await fetchEmergencyPatient(id);
          setPatient(data);
        }
      } catch (error) {
        console.error("Error fetching patient details:", error);
      }
    };

    loadPatient();
  }, [id]);

  if (!patient) {
    return <div>Loading patient details...</div>;
  }

  return (
    <>
      <h1 className="patient-details-title">Patient Details</h1>
      <div className="patient-details-container">
        <p>
          <strong>ID:</strong> {patient.patientId}
        </p>
        <p>
          <strong>Given Names:</strong> {patient.givenNames}
        </p>
        <p>
          <strong>Family Name:</strong> {patient.familyName}
        </p>
        <button
          className="back-button"
          onClick={() => navigate("/emergency-ward")}
        >
          Back
        </button>
      </div>
    </>
  );
};

export default PatientDetails;
