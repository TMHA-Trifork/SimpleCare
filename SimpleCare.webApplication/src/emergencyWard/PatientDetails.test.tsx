import { describe, it, expect, vi, beforeEach, afterEach, Mock } from "vitest";
import { createRoot } from "react-dom/client";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import PatientDetails from "./PatientDetails";
import * as api from "../api/emergencyPatients";

vi.mock("../api/emergencyPatients");

const mockFetchEmergencyPatient = api.fetchEmergencyPatient as Mock;

describe("PatientDetails", () => {
  let container: HTMLDivElement;
  let root: ReturnType<typeof createRoot>;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
    root = createRoot(container);
  });

  afterEach(() => {
    root.unmount();
    document.body.removeChild(container);
    vi.clearAllMocks();
  });

  it("should match snapshot", async () => {
    mockFetchEmergencyPatient.mockResolvedValueOnce({
      patientId: "1",
      givenNames: "John",
      familyName: "Doe",
    });

    root.render(
      <MemoryRouter initialEntries={["/patient/1"]}>
        <Routes>
          <Route path="/patient/:id" element={<PatientDetails />} />
        </Routes>
      </MemoryRouter>
    );

    await vi.waitFor(() => {
      expect(container.textContent).toContain("John");
      expect(container.textContent).toContain("Doe");
    });

    await vi.waitFor(() => {
      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  it("displays patient details after loading", async () => {
    mockFetchEmergencyPatient.mockResolvedValueOnce({
      patientId: "1",
      givenNames: "John",
      familyName: "Doe",
    });

    root.render(
      <MemoryRouter initialEntries={["/patient/1"]}>
        <Routes>
          <Route path="/patient/:id" element={<PatientDetails />} />
        </Routes>
      </MemoryRouter>
    );

    await vi.waitFor(() => {
      expect(container.textContent).toContain("John");
      expect(container.textContent).toContain("Doe");
    });
  });
});
