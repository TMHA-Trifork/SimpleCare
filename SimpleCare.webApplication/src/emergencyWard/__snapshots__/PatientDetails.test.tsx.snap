// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PatientDetails > should match snapshot 1`] = `"<h1 class="patient-details-title">Patient Details</h1><div class="patient-details-container"><p><strong>ID:</strong> 1</p><p><strong>Given Names:</strong> John</p><p><strong>Family Name:</strong> Doe</p><button class="back-button">Back</button></div>"`;
